shared/templates/create_permissions.py

```diff
@@ -0,0 +1,220 @@
+
+            # we are ready to create the check
+            # open the template and perform the conversions
+            if mode:
+                # build the state that describes our mode
+                # mode_str maps to STATEMODE in the template
+                fields = ['oexec', 'owrite', 'oread', 'gexec', 'gwrite', 'gread',
+                          'uexec', 'uwrite', 'uread', 'sticky', 'sgid', 'suid']
+                mode_int = int(mode, 8)
+                mode_str = ""
+                for field in fields:
+                    if mode_int & 0x01 == 1:
+                        mode_str = "	<unix:" + field + " datatype=\"boolean\">true</unix:" + field + ">\n" + mode_str
+                    else:
+                        mode_str = "	<unix:" + field + " datatype=\"boolean\">false</unix:" + field + ">\n" + mode_str
+                    mode_int = mode_int >> 1
+
+                self.file_from_template(
+                    "./template_OVAL_file_permissions",
+                    {
+                        "FILEID":        path_id,
+                        "FILEPATH":      full_path,
+                        "FILEMODE":      mode,
+                        "STATEMODE":     mode_str,
+                        "UNIX_DIR":      unix_dir,
+                        "UNIX_FILENAME": unix_filename
+                    },
+                    "./oval/file_permissions{0}.xml", path_id
+                )
+            if uid:
+                self.file_from_template(
+                    "./template_OVAL_file_owner",
+                    {
+                        "FILEID":        path_id,
+                        "FILEPATH":      full_path,
+                        "FILEUID":       uid,
+                        "UNIX_DIR":      unix_dir,
+                        "UNIX_FILENAME": unix_filename
+                    },
+                    "./oval/file_owner{0}.xml", path_id
+                )
+            if gid:
+                self.file_from_template(
+                    "./template_OVAL_file_groupowner",
+                    {
+                        "FILEID":        path_id,
+                        "FILEPATH":      full_path,
+                        "FILEGID":       gid,
+                        "UNIX_DIR":      unix_dir,
+                        "UNIX_FILENAME": unix_filename
+                    },
+                    "./oval/file_groupowner{0}.xml", path_id
+                )
+            self.file_from_template(
+                "./template_OVAL_permissions",
+                {
+                    "FILEID":        path_id,
+                    "FILEPATH":      full_path,
+                    "FILEDIR":       dir_path,
+                    "FILEMODE":      mode,
+                    "FILEUID":       uid,
+                    "FILEGID":       gid,
+                    "UNIX_DIR":      unix_dir,
+                    "UNIX_FILENAME": unix_filename
+                },
+                "./oval/permissions{0}.xml", path_id
```

Comment:
I couldn't find any rule named with prefix `permissions_`... Will this be picked up by build system?

Reply:
So... I'm not sure here either, but this was the original behavior. If we want rules that begin with `permissions_`, we should have the corresponding rules in the `linux_os` tree and they should be referenced as such in the correct profiles. As it stands, it looks like there are no such rules.